nixos/modules/flyingcircus/roles/statshost.nix

```diff
@@ -262,7 +270,39 @@ in
           }];
           metric_relabel_configs = cfgStatsGlobal.prometheusMetricRelabel;
         }
-      ];
+
+      ] ++ relayConfig;
+
+      # Update relayed nodes.
+      systemd.services.fc-prometheus-update-relayed-nodes = {
+        description = "Update prometheus proxy relayed nodes.";
+        restartIfChanged = false;
+        after = [ "network.target" ];
+        wantedBy = [ "prometheus.service" ];
+        serviceConfig = {
+          User = "root";
+          Type = "oneshot";
+        };
+        path = [ pkgs.curl pkgs.coreutils ];
+        script = concatStringsSep "\n" (map
+          (relayNode: ''
+            curl -v -o /etc/local/statshost/relay-${relayNode.job_name}.json \
+              ${relayNode.proxy_url}/scrapeconfig.json
```

Comment:
I'd put self-updating stuff into /var/cache or /var/lib, not into /etc/. The assumption is that the administrator/service user can change a file in /etc/local and it stays the same.